### ISSUE TITLE
Integrate subsmod

### ISSUE THREAD

**Issue by BlackPhrase** [BlackPhrase/magenta#41]
@Fograin , you did a great job
And it won't be lost or forgotten
I will integrate your changes right into the engine
(If I stay motivated by that time...)

**Comment by Fograin** [comment]
@BlackPhrase , Thank you :D 
Feel free to use any part of subsmod, it would be great to see that subs code won't be lost and forgotten :)

**Comment by BlackPhrase** [comment]
Thanks!

I've read the [update_license.txt](https://github.com/Fograin/hl-subsmod-ex/blob/master/update_license.txt)
I think I should clarify some things

First, I'm reimplementing the game code from original QC progs code
I think you know why
The reason is pretty much the same why the subsmod was shut down (because I'm unable to plug the original GS mods with the reimplemented engine without violating the HLSDK EULA)
Is that ok?

Second, what is the license for this folder contents?
https://github.com/Fograin/hl-subsmod-ex/tree/master/mod_src/subs_mod

And third, does the subsmod sources contains some reversed parts (excluding the Xash sources)?
I mean, are the main subsmod sources clean from the legal perspective? I'm planning to disconnect the mod from HLSDK and Xash engine and I need to be sure that the remaining part is clean

**Comment by Fograin** [comment]
First; That's awesome :D It's the best solution to grab QC code and build something that's fully legal with QC/HLSDK codes. However you need to think about game code availability. If you want to do full open source project then everything should be OK with both Valve and ID Software. If you want to keep source code private but release game for free then it also should be OK. I guess Valve/ID could only give you some trouble if you would try to sell that stuff ;)

Second: Ahh the mighty subtitles system folder :D It's based on CC 3.0, basically you can do whatever you want with it copy/edit/redistribute but you need to acknowledge autors. The whole system was written by guy named "Vitamin" (I Don't know his real name" and then I made few edits. Just put stuff from HL: Update mod credits into your game/mod credits and everything will be OK :)

Third: Well excluding Xash sources then... there shouldn't be anything illegal there. Basically the whole Subs Mod/HLU: Was made by using Xash engine + HLSDK. The whole "mod" code was created using client/server side projects, so It's fully based on HLSDK 2.3. However there are a few changes in Xash engine. For example game/mod load/mounting options. In order to "mount" valve, gearbox, bshift and subsmod directories I've changed loader in engine file to mount everything in order. And I think I've edited some engine limits (increased limits of skins, submodels and max triangles per model).

In short almost everything from subsmod-ex is based on HLSDK client/server and implemented as mod, but you might need to tweak some QC engine code for them to work.

**Comment by BlackPhrase** [comment]
Well, for game implementation I'm using the original Quake's QC code ported to C++ as a basis
That's the same way Valve did for Half-Life
And I'm not using the original HLSDK because of its stupid EULA
I've partially reimplemented the HLSDK pieces (mostly engine structures and some interfaces) using the original Quake code
The whole codebase is GPL3 so everything is open-source
